main.go

```diff
@@ -235,10 +228,28 @@ func sendToWriter(ingestionType string, msgBody io.Reader, uuid string, URLSlice
 	if err != nil {
 		log.Errorf("Cannot read error body: [%v]", err)
 	}
-
 	return fmt.Errorf("reqURL=[%s] status=[%d] uuid=[%s] error=[%v] body=[%s]", reqURL, resp.StatusCode, uuid, reqErr, string(errorMessage))
 }
 
+func resolveWriterAndCreateRequest(ingestionType string, msgBody io.Reader, uuid string, URLSlice []string) (*http.Request, string, error) {
+	var writerURL string
+	for _, URL := range URLSlice {
```

Comment:
Can the URLs be mapped to the service/ingestionType? It would simplify this lookup too.